### ISSUE TITLE
Make package manager configuration more generic

### ISSUE THREAD

**Issue by DaanDeMeyer** [systemd/mkosi#1603]
There's stuff like package pinning and tons of other configuration that's very package manager specific and wouldn't make sense for mkosi to implement for each package manager. Instead, we can allow users to specify and "etc" directory for the package manager which we treat as /etc for the package manager specifically. Users can put whatever package manager config files in there that they want and we'll make sure those files and directories are taken into account by the package manager.

**Comment by behrmann** [comment]
As discussed previously this overlaps skeletons quite strongly (though not fully, since e.g. postinst scripts in Debian may react to files already in a root), so a solution I had in mind would be to copy the skeletons both into the image and into a separate dir in the workspace and resolve all config relative to that root.

**Comment by DaanDeMeyer** [comment]
The big difference is that skeleton trees are copied into the image, and this directory would not be.

**Comment by behrmann** [comment]
That's true, but is that necessary, though? I see two scenarios
1. You wanted an image where the package manager can run after you deploy it, then you will probably want to use the same config (repos, allowed keys, package pins) for the package manager in the image as when generating it
2. You don't want to run the package manager in the image, but then you don't really care whether these files will be copied into the image, because you either don't care they are there or you clean them up already.

**Comment by gdiscry** [comment]
Like @behrmann, I only see those two scenarios.

Personally, I mainly build Debian images matching scenario 1 (but I am interested in building Arch images matching scenario 2). With the previous releases of mkosi (14 and before), I used `mkosi.skeleton` to add the backports repository, tweak the package selection in `/etc/apt/preferences.d` and seed a debconf configuration in `/var/cache/debconf`. I also disabled CleanPackageMetadata.

With the coming version, I understand the shift towards running the package manager directly on the host. I was glad to see debootstrap go as it does not support a skeleton filesystem, which is a pain with debconf. However, moving the configuration outside the image meant that configuring APT does not work anymore, hence the PRs I sent some time ago to start fixing that, starting with adding custom repositories.

So here are my 2 cents:

* I get that configuring the package manager is complicated because mkosi provides parts of the configuration and the user might want to tweak the rest, but `mkosi.skeleton` fits that role as long as mkosi can inject/override what it needs.
* There might also be some differences between the configuration used when building the image and the configuration inside the final image (like the use of a local mirror for custom repositories), which can be covered by `mkosi.extra`.
* Finally, I also think that scenario 2 is better served by cleaning up the image at the end. Is there even a distribution where that scenario is possible without cleaning up?

I'm glad to see that this ticket was opened. I think that a few use cases where a new tree is more useful than `mkosi.skeleton` + `mkosi.extra` + clean up could help define what's needed.

P.S.: Maybe the clean up could be done by CleanPackageMetadata or something similar.
